test/Worker.Extensions.Tests/AspNetIntegration/RegistrationExpectedInAspNetIntegrationTests.cs

```diff
@@ -0,0 +1,341 @@
+                    using Microsoft.Extensions.Hosting;
+                    using Microsoft.Extensions.Logging;
+
+                    class Program
+                    {
+                        static void Main(string[] args)
+                        {
+                            var host = new HostBuilder()
+                                .ConfigureFunctionsWorkerDefaults()
+                                .Build();
+
+                            host.Run();
+                        }
+                    }
+                }
+
+                namespace AspNetIntegration
+                {
+                    using Microsoft.Azure.Functions.Worker;
+                    using Microsoft.Azure.Functions.Worker.Http;
+
+                    public class FunctionHttpTrigger
+                    {
+                        [Function(nameof(FunctionHttpTrigger))]
+                        public void Run([HttpTrigger(AuthorizationLevel.Anonymous, ""post"")] HttpRequestData req)
+                        {
+                        }
+                    }
+                }";
+
+            var test = new AnalyzerTest
+            {
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net60.WithPackages(ImmutableArray.Create(
+                    new PackageIdentity("Microsoft.Azure.Functions.Worker", "1.19.0"),
+                    new PackageIdentity("Microsoft.Azure.Functions.Worker.Sdk", "1.14.1"),
+                    new PackageIdentity("Microsoft.Azure.Functions.Worker.Extensions.Storage.Blobs", "6.0.0"),
+                    new PackageIdentity("Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore", "1.0.0"),
+                    new PackageIdentity("Microsoft.Azure.Functions.Worker.Extensions.Abstractions", "5.0.0"),
+                    new PackageIdentity("Microsoft.Extensions.Hosting.Abstractions", "6.0.0"),
+                    new PackageIdentity("Microsoft.Azure.Functions.Worker.Extensions.Http", "3.1.0")
+                    )),
+
+                TestCode = testCode
+            };
+
+            test.ExpectedDiagnostics.Add(Verifier.Diagnostic()
+                            .WithSeverity(Microsoft.CodeAnalysis.DiagnosticSeverity.Error)
+                            .WithSpan(13, 37, 13, 41)
+                            .WithArguments(ExpectedRegistrationMethod));
+
+            await test.RunAsync();
+        }
+
+        [Fact]
+        public async Task NotAspNetIntegration_Diagnostics_NotExpected()
+        {
+            string testCode = @"
+                using System;
+                using Microsoft.Extensions.DependencyInjection;
+                using Microsoft.Extensions.Hosting;
+
+                namespace SampleApp
+                {
+                    public class Program
+                    {
+                        public static void Main()
+                        {
+                            var host = new HostBuilder()
+                                .ConfigureFunctionsWorkerDefaults()
```

Comment:
What happens if someone calls `ConfigureFunctionsWorkerDefaults` followed by `ConfigureFunctionsWebApplication` ?

Reply:
```csharp
var host = new HostBuilder()
    .ConfigureFunctionsWorkerDefaults()
    .ConfigureFunctionsWebApplication()
    .Build();
```

Reply:
yeah, tried above scenario with an aspnet function app and it did not work. Didn't get proper exception also. Do we have an issue tracking to fix this behavior? For analyzer, I can add a new one for this scenario to keep the [error code](https://github.com/Azure/azure-functions-dotnet-worker/tree/main/docs/analyzer-rules) specific to one issue.

Reply:
Update - had an offline sync on this. The investigation for this scenario will be tracked [here](https://github.com/Azure/azure-functions-dotnet-worker/issues/1947)